src/pool/rpc.h

```diff
@@ -208,7 +211,9 @@ CRT_RPC_DECLARE(pool_disconnect, DAOS_ISEQ_POOL_DISCONNECT,
 	((struct daos_pool_space) (pqo_space)		CRT_RAW) \
 	((struct daos_rebuild_status) (pqo_rebuild_st)	CRT_RAW) \
 	/* only set on -DER_TRUNC */				 \
-	((uint32_t)		(pqo_map_buf_size)	CRT_VAR)
+	((uint32_t)		(pqo_map_buf_size)	CRT_VAR) \
+	((uint32_t)		(pqo_pool_layout_ver)	CRT_VAR) \
```

Comment:
need to remove the additions to this RPC definition since the patch is no longer going to modify the pool query cart RPC and API definition?

Reply:
And this is needed in ds_mgmt_drpc_pool_query() as pool list will issue pool query.

Reply:
Yes OK right. The new protocol is effectively used only on the engine side with this patch. And clients, while they may dlopen a libdaos that speaks this new rpc protocol, the pool query API will not be changed in this patch (and so clients don't need to be rebuilt).

I think I was getting confused about another idea I had, to possibly have the engine retrieve the layout_ver by querying a pool property.

Reply:
Isn't this a protocol change?   Does it require any handshake (like we do with key query) to figure out what version the server supports?   It's possible to use a new client with an older engine

Reply:
See dc_obj_init()